pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -273,6 +273,34 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		Status: operatorv1.ConditionFalse,
 	}
 
+	if !preconditionsReady {
+		message := "the operator haven't specified what preconditions are missing"
+		for _, err := range errs {
+			message = message + err.Error() + "\n"
+		}
```

Comment:
message = strings.Join(errs, "\n")

Reply:
string.Joins accepts `[]string`, won't accept `errs` which is `error`

`string.Join(elems []string, sep string) string`

Reply:
:facepalm: I did not realize we're dealing with errors, sorry, my bad